transformer_engine/common/fused_attn/fused_attn.cpp

```diff
@@ -80,7 +75,8 @@ NVTE_Fused_Attn_Backend nvte_get_fused_attn_backend(
         NVTE_Bias_Type bias_type,
         NVTE_Mask_Type attn_mask_type,
         float dropout, size_t max_seqlen_q,
-        size_t max_seqlen_kv, size_t head_dim) {
+        size_t max_seqlen_kv, size_t head_dim,
+        size_t num_attn_heads, size_t num_gqa_groups) {
```

Comment:
I was wondering if we can keep the order of these arguments aligned for all fused attention API
I noticed there are a few different orders:

1. get_fused_attn_backend: s, d, h, hg
2. max512, arbitrary_qkvpack: b, s, h, d
3. transformer_engine::fused_attn::arbitrary: b, h, hg, s, d
4. arbitrary, arbitrar_kvpack: b, s, h, hg, d

I preferred to use `b, s, h, hg, d` as the order of these arguments and skip one of them if it is not necessary. i.e. `s, h, hg, d` for `get_fused_attn_backend`; `b, s, h, d` for max512

Reply:
I will have to clean this up a bit.

Reply:
Done